web/frontend/src/components/slam-2d-render.vue

```diff
@@ -78,6 +108,21 @@ const disposeScene = () => {
   scene.clear();
 };
 
+// Find the desired color bucket for a given probability. This assumes the probability will be a value from 0 to 100
+const probToColorMapBucket = (normProb: number, numBuckets: number): number => {
+  const prob = Math.max(Math.min(100, normProb * 255), 0);
+  return Math.floor((numBuckets - 1) * prob / 100);
+};
+
+/*
+ * Map the color of a pixel to a color bucket value.
+ * normProb is the probability value normalized by the size of a byte(255) to be between 0 to 1.
+ */
+const colorBuckets = (normProb: number): THREE.Vector3 => {
+  return colorMapGrey.map(([red, green, blue]) =>
+    new THREE.Vector3(red, green, blue).multiplyScalar(1 / 255))[probToColorMapBucket(normProb, colorMapGrey.length)]!;
```

Comment:
What does the `.multiplyScalar(1 / 255)` do?

Reply:
this normalizes the pixels of `colorMapGrey`. So the vector3 will contain values of `0 to 1` instead of `0 to 255`, which would match the expected values that `colors` wants

Reply:
GPUs usually like data normalized to 0-1 :)